src/Pools/Pool.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace Utopia\Pools;
+
+use Exception;
+
+class Pool
+{
+    /**
+     * @var string
+     */
+    protected string $name;
+    
+    /**
+     * @var int
+     */
+    protected int $size = 0;
+
+    /**
+     * @var callable
+     */
+    protected $init = null;
+
+    /**
+     * @var int
+     */
+    protected int $reconnectAttempts = 3;
+
+    /**
+     * @var int
+     */
+    protected int $reconnectSleep = 1; // seconds
+
+    /**
+     * @var array
+     */
+    protected array $pool = [];
+
+    /**
+     * @var array
+     */
+    protected array $active = [];
+
+    /**
+     * @var string $name
+     * @var int $size
+     * @var callable $init
+     */
+    public function __construct(string $name, int $size, callable $init)
+    {
+        $this->name = $name;
+        $this->size = $size;
+        $this->init = $init;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSize(): int
+    {
+        return $this->size;
+    }
+
+    /**
+     * @return int
+     */
+    public function getReconnectAttempts(): int
+    {
+        return $this->reconnectAttempts;
+    }
+
+    /**
+     * @var int $reconnectAttempts
+     * @return self
+     */
+    public function setReconnectAttempts(int $reconnectAttempts): self
+    {
+        $this->reconnectAttempts = $reconnectAttempts;
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getReconnectSleep(): int
+    {
+        return $this->reconnectSleep;
+    }
+
+    /**
+     * @var int $reconnectSleep
+     * @return self
+     */
+    public function setReconnectSleep(int $reconnectSleep): self
+    {
+        $this->reconnectSleep = $reconnectSleep;
+        return $this;
+    }
+
+    /**
+     * @return self
+     */
+    public function fill(): self
+    {
+        $this->pool = [];
+
+        for ($i=0; $i < $this->size; $i++) { 
+            $attempts = 0;
+
+            do {
+                try {
+                    $attempts++;
+                    $connection = new Connection(($this->init)());
```

Comment:
Is there a difference b/w `new Connection(($this->init)());` and `new Connection($this->init());`

Reply:
Nope, probably copilot autocomplete